pkg/drivers/vfkit/driver_darwin.go

```diff
@@ -248,6 +254,20 @@ func (d *Driver) Start() error {
 		return err
 	}
 
+	// when loading a VM created by a crc version predating this commit,
+	// d.VsockQemuGAPort will be missing from ~/.crc/machines/crc/config.json
+	// In such a case, assume the VM will not support time sync
+	if d.VsockQemuGAPort != 0 {
+		timesync, err := client.TimeSyncNew(d.VsockQemuGAPort)
```

Comment:
the device we add is not really timesync, but the guest agent. timesyncing is just one (only) of the features used.

Reply:
it's more 'timesync' than just guest agent. The guest agent connection is not exposed externally, it's a net.Conn instance internal to vfkit. And vfkit monitors for wake up events, and sends a `set-time $current_time` command to the guest agent if present.

Reply:
If that is the case, naming should represent that; `GuestAgentForTimesync` in some form inside the codebase.

Reply:
Or just `VsockQemuGAPort` -> `VsockTimesyncPort`?